### DISCUSSION TITLE
How to optimize disk consumption?

### DISCUSSION THREAD

**Discussion by afathonih** [lancedb/lancedb#817] [Q&A]
Hi,
I'm doing a PoC using LanceDB to index source code. At the moment, it already consumed 84GB. 
`du` shows:
```
 82G    lancedb/my_app.lance/_versions
214M    lancedb/my_app.lance/_transactions
428M    lancedb/my_app.lance/data
 83G    lancedb/my_app.lance
 84G    lancedb
```

Is there a way to optimize the disk usage? Is it safe to drop `_version`? I don't need it at this stage.

Thank you for this cool product, by the way!

**Comment by ChenZhongPu** [comment]
I have the same doubt. For read-only task, is it safe to discard `_versions` and `_transactions`?

**Reply by wjones127** [reply]
Do not delete `_versions`! That would make your table unreadable. Instead use `cleanup_old_versions` as described by Weston to delete the files no longer in use.

**Comment by westonpace** [comment]
We keep old data around for time-travel / versioning purposes.  You can use `cleanup_old_versions` to safely remove old data (once you do this you will not be able to `checkout` these old versions).

Unfortunately, it seems this is only exposed via python at the moment.  I'll add it to nodejs soon.

Python: https://lancedb.github.io/lancedb/python/python/#lancedb.table.Table.update

**Comment by SommerEngineering** [comment]
Thank you, @afathonih, for asking this question. The function `cleanup_old_versions` doesn't seem to exist in Rust either. Is there a workaround for Rust projects? I can't integrate Python code into my project... 🙁

**Reply by wjones127** [reply]
The `optimize` method is what you are looking for: https://docs.rs/lancedb/latest/lancedb/table/struct.Table.html#method.optimize
